extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +148,223 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!Window} win The window element
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(win, transportOptions, opt_processResponse) {
+    const frameUrl = dev().assertString(transportOptions['iframe'],
+        'Cross-domain frame parameters missing ${this.type_}');
+    let frameData;
+    let shouldAddToDOM = false;
+    if (Transport.hasCrossDomainIframe(frameUrl)) {
+      frameData = Transport.getFrameData(frameUrl);
+      Transport.incrementIframeUsageCount_(
+        /** @type{!FrameData} */ (frameData));
+    } else {
+      frameData = this.createCrossDomainIframe(win, frameUrl,
+          opt_processResponse);
+      shouldAddToDOM = true;
+    }
+    dev().assert(frameData, 'Trying to use non-existent frame');
+    if (transportOptions['extraData']) {
+      frameData.queue.setExtraData(this.id_, transportOptions['extraData']);
+    }
+    if (shouldAddToDOM) {
+      win.document.body.appendChild(frameData.frame);
+    }
+  }
+
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!Window} win The window element
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   * @param {function(!string,
+   *   !../../../src/3p-analytics-common.AmpAnalytics3pResponse)=}
+   *   opt_processResponse An optional function to receive any response
+   *   messages back from the cross-domain iframe
+   * @return {!FrameData}
+   * @VisibleForTesting
+   */
+  createCrossDomainIframe(win, frameUrl, opt_processResponse) {
+    const sentinel = Transport.createUniqueId_();
+    const scriptSrc = getMode().localDev
```

Comment:
@keithwrightbos I changed to use calculateEntryPointScriptUrl() here, as @lannka suggested. However, in the other PR (https://github.com/google/amphtml/pull/20/files -> last few lines of examples/analytics-3p-remote-frame.html) that brings us back to:
```
script.src = JSON.parse(window.name).scriptSrc;
```
You had previously expressed concern about XSS when receiving the _entire_ URL. But of course we can't use calculateEntryPointScriptUrl() in analytics-3p-remote-frame.html since that would introduce a chicken-and-egg scenario.
Thoughts?